metadata-ingestion/src/datahub/ingestion/source/powerbi.py

```diff
@@ -104,41 +104,45 @@ class Constant:
     VALUE = "value"
     ENTITY = "ENTITY"
     ID = "ID"
+    HTTP_RESPONSE_TEXT = "HttpResponseText"
+    HTTP_RESPONSE_STATUS_CODE = "HttpResponseStatusCode"
+
+
+@dataclass
+class CapabilitiesOverride:
+    extractOwnership: bool = pydantic.Field(
+        default=True,
+        description="Enable / Disable ingestion of user information for dashboard owner",
+    )
 
 
 class PowerBiAPIConfig(EnvBasedSourceConfigBase):
     # Organsation Identifier
-    tenant_id: str = Field(description="Power BI tenant identifier.")
+    tenant_id: str = pydantic.Field(description="PowerBI tenant identifier")
     # PowerBi workspace identifier
-    workspace_id: str = Field(description="Power BI workspace identifier.")
+    workspace_id: str = pydantic.Field(description="PowerBI workspace identifier")
     # Dataset type mapping
-    dataset_type_mapping: Dict[str, str] = Field(
-        description="Mapping of Power BI datasource type to Datahub dataset."
+    dataset_type_mapping: Dict[str, str] = pydantic.Field(
+        description="Mapping of PowerBI datasource type to DataHub supported data-sources. See Quickstart Recipe for mapping"
     )
     # Azure app client identifier
-    client_id: str = Field(description="Azure AD App client identifier.")
+    client_id: str = pydantic.Field(description="Azure app client identifier")
     # Azure app client secret
-    client_secret: str = Field(description="Azure AD App client secret.")
+    client_secret: str = pydantic.Field(description="Azure app client secret")
     # timeout for meta-data scanning
-    scan_timeout: int = Field(
-        default=60,
-        description="time in seconds to wait for Power BI metadata scan result.",
+    scan_timeout: int = pydantic.Field(
+        default=60, description="timeout for PowerBI metadata scanning"
     )
-
-    scope: str = "https://analysis.windows.net/powerbi/api/.default"
-    base_url: str = "https://api.powerbi.com/v1.0/myorg/groups"
-    admin_base_url = "https://api.powerbi.com/v1.0/myorg/admin"
-    authority = "https://login.microsoftonline.com/"
-
-    def get_authority_url(self):
-        return f"{self.authority}{self.tenant_id}"
+    # Enable/Disable extracting ownership information of Dashboard
+    override: CapabilitiesOverride = CapabilitiesOverride(extractOwnership=True)
```

Comment:
I'm in favor of not introducing another level of nesting into the config. We have this pattern of bool flags throughout the codebase, for example the `include_views` option on the SQL sources. In those cases, it'd be pretty confusing to have `include_views` nested under a capabilities object but have the `view_patterns` still be top-level. Overall, I think we should keep the config flat unless there's a compelling reason not to.